actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +364,14 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.class.in? [ActiveStorage::Variant, ActiveStorage::Blob, ActiveStorage::Attachment]
```

Comment:
Ha! That's how I was going to do it, but thought it will be rejected as too far-fetched. E.g. "if someone goes as far as to write their own image framework with their own custom routes, they surely can override image_tag themselves".